### DISCUSSION TITLE
Add Boost as a dependency?

### DISCUSSION THREAD

**Discussion by vlolteanu** [diasurgical/DevilutionX#1752] [General]
Using Boost trivial on Linux, but I have no idea how hard it is to use it on some of the other platforms that DevilutionX supports.

There are a few places where stuff provided by Boost fits nicely:
 * shared resource ownership: I've noticed a couple of places where boost::container::local_shared_ptr might come in handy (shared_ptr does basically the same, but employs atomic variables, which are unnecessary here)
 * boost::container::static_vector is roughly a 1:1 container replacement for fixed-size arrays with a member counter (i.e. SomeStruct something[MAX_SOMETHING]; int somethingCount). This pattern is very prevalent.

I'm sure that if I look further, I can find more use cases.

Would it be a big deal to require Boost?

**Comment by AJenbo** [comment]
Boost is too aggressive of a dependency

**Reply by vlolteanu** [reply]
Well, I see you've brought in boost::asio :)

Would individual files be ok, in principle?  I know it would come down to the specifics of every file.

**Reply by AJenbo** [reply]
yeah, but its optional, and we would like to replace it with SDL_net.

**Reply by AJenbo** [reply]
possibly, if the rational is there and the content reasonable for what it does
